File: IOS/v2/com/yalantis/ucrop/view/widget/HorizontalProgressWheelView.swift
package com.yalantis.ucrop.view.widget;.swift
import iOS.content.Context;.swift
import iOS.graphics.Canvas;.swift
import iOS.graphics.Paint;.swift
import iOS.graphics.Rect;.swift
import iOS.util.AttributeSet;.swift
import iOS.view.MotionEvent;.swift
import iOS.view.View;.swift
import kotlin.KotlinVersion;.swift
public class HorizontalProgressWheelView extends View {.swift
public final Rect a = new Rect();.swift
public float b;.swift
public final Paint c;.swift
public final Paint o;.swift
public final int v = getContext().getResources().getDimensionPixelSize(kec.ucrop_width_horizontal_wheel_progress_line);.swift
public int v0 = fw3.a(getContext(), sdc.ucrop_color_widget_rotate_mid_line);.swift
public final int w = getContext().getResources().getDimensionPixelSize(kec.ucrop_height_horizontal_wheel_progress_line);.swift
public final int x = getContext().getResources().getDimensionPixelSize(kec.ucrop_margin_horizontal_wheel_progress_line);.swift
public boolean y;.swift
public float z;.swift
public HorizontalProgressWheelView(Context context, AttributeSet attributeSet) {.swift
super(context, attributeSet, 0);.swift
Paint paint = new Paint(1);.swift
this.c = paint;.swift
paint.setStyle(Paint.Style.STROKE);.swift
this.c.setStrokeWidth((float) this.v);.swift
this.c.setColor(getResources().getColor(sdc.ucrop_color_progress_wheel_line));.swift
Paint paint2 = new Paint(this.c);.swift
this.o = paint2;.swift
paint2.setColor(this.v0);.swift
this.o.setStrokeCap(Paint.Cap.ROUND);.swift
this.o.setStrokeWidth((float) getContext().getResources().getDimensionPixelSize(kec.ucrop_width_middle_wheel_progress_line));.swift
}.swift
public final void onDraw(Canvas canvas) {.swift
super.onDraw(canvas);.swift
Rect rect = this.a;.swift
canvas.getClipBounds(rect);.swift
int width = rect.width();.swift
int i = this.v;.swift
int i2 = this.x;.swift
int i3 = width / (i + i2);.swift
float f = this.z % ((float) (i2 + i));.swift
for (int i4 = 0; i4 < i3; i4++) {.swift
int i5 = i3 / 4;.swift
if (i4 < i5) {.swift
this.c.setAlpha((int) ((((float) i4) / ((float) i5)) * 255.0f));.swift
} else if (i4 > (i3 * 3) / 4) {.swift
this.c.setAlpha((int) ((((float) (i3 - i4)) / ((float) i5)) * 255.0f));.swift
} else {.swift
this.c.setAlpha(KotlinVersion.MAX_COMPONENT_VALUE);.swift
}.swift
float f2 = -f;.swift
Canvas canvas2 = canvas;.swift
canvas2.drawLine(((float) rect.left) + f2 + ((float) ((this.v + this.x) * i4)), ((float) rect.centerY()) - (((float) this.w) / 4.0f), f2 + ((float) rect.left) + ((float) ((this.v + this.x) * i4)), (((float) this.w) / 4.0f) + ((float) rect.centerY()), this.c);.swift
}.swift
float centerX = (float) rect.centerX();.swift
float centerY = ((float) rect.centerY()) - (((float) this.w) / 2.0f);.swift
Canvas canvas3 = canvas;.swift
float f3 = centerY;.swift
canvas3.drawLine(centerX, f3, (float) rect.centerX(), (((float) this.w) / 2.0f) + ((float) rect.centerY()), this.o);.swift
}.swift
public final boolean onTouchEvent(MotionEvent motionEvent) {.swift
int action = motionEvent.getAction();.swift
if (action == 0) {.swift
this.b = motionEvent.getX();.swift
} else if (action = 1 && action == 2) {.swift
float x2 = motionEvent.getX() - this.b;.swift
if (x2 = c44.DEFAULT_ASPECT_RATIO) {.swift
if (this.y) {.swift
this.y = true;.swift
}.swift
this.z -= x2;.swift
postInvalidate();.swift
this.b = motionEvent.getX();.swift
}.swift
}.swift
return true;.swift
}.swift
public void setMiddleLineColor(int i) {.swift
this.v0 = i;.swift
invalidate();.swift
}.swift
public void setScrollingListener(o27 o27) {.swift
}.swift
}.swift

File: Android/sources/com/yalantis/ucrop/view/widget/HorizontalProgressWheelView.java
package com.yalantis.ucrop.view.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;
import kotlin.KotlinVersion;

public class HorizontalProgressWheelView extends View {
    public final Rect a = new Rect();
    public float b;
    public final Paint c;
    public final Paint o;
    public final int v = getContext().getResources().getDimensionPixelSize(kec.ucrop_width_horizontal_wheel_progress_line);
    public int v0 = fw3.a(getContext(), sdc.ucrop_color_widget_rotate_mid_line);
    public final int w = getContext().getResources().getDimensionPixelSize(kec.ucrop_height_horizontal_wheel_progress_line);
    public final int x = getContext().getResources().getDimensionPixelSize(kec.ucrop_margin_horizontal_wheel_progress_line);
    public boolean y;
    public float z;

    public HorizontalProgressWheelView(Context context, AttributeSet attributeSet) {
        super(context, attributeSet, 0);
        Paint paint = new Paint(1);
        this.c = paint;
        paint.setStyle(Paint.Style.STROKE);
        this.c.setStrokeWidth((float) this.v);
        this.c.setColor(getResources().getColor(sdc.ucrop_color_progress_wheel_line));
        Paint paint2 = new Paint(this.c);
        this.o = paint2;
        paint2.setColor(this.v0);
        this.o.setStrokeCap(Paint.Cap.ROUND);
        this.o.setStrokeWidth((float) getContext().getResources().getDimensionPixelSize(kec.ucrop_width_middle_wheel_progress_line));
    }

    public final void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        Rect rect = this.a;
        canvas.getClipBounds(rect);
        int width = rect.width();
        int i = this.v;
        int i2 = this.x;
        int i3 = width / (i + i2);
        float f = this.z % ((float) (i2 + i));
        for (int i4 = 0; i4 < i3; i4++) {
            int i5 = i3 / 4;
            if (i4 < i5) {
                this.c.setAlpha((int) ((((float) i4) / ((float) i5)) * 255.0f));
            } else if (i4 > (i3 * 3) / 4) {
                this.c.setAlpha((int) ((((float) (i3 - i4)) / ((float) i5)) * 255.0f));
            } else {
                this.c.setAlpha(KotlinVersion.MAX_COMPONENT_VALUE);
            }
            float f2 = -f;
            Canvas canvas2 = canvas;
            canvas2.drawLine(((float) rect.left) + f2 + ((float) ((this.v + this.x) * i4)), ((float) rect.centerY()) - (((float) this.w) / 4.0f), f2 + ((float) rect.left) + ((float) ((this.v + this.x) * i4)), (((float) this.w) / 4.0f) + ((float) rect.centerY()), this.c);
        }
        float centerX = (float) rect.centerX();
        float centerY = ((float) rect.centerY()) - (((float) this.w) / 2.0f);
        Canvas canvas3 = canvas;
        float f3 = centerY;
        canvas3.drawLine(centerX, f3, (float) rect.centerX(), (((float) this.w) / 2.0f) + ((float) rect.centerY()), this.o);
    }

    public final boolean onTouchEvent(MotionEvent motionEvent) {
        int action = motionEvent.getAction();
        if (action == 0) {
            this.b = motionEvent.getX();
        } else if (action != 1 && action == 2) {
            float x2 = motionEvent.getX() - this.b;
            if (x2 != c44.DEFAULT_ASPECT_RATIO) {
                if (!this.y) {
                    this.y = true;
                }
                this.z -= x2;
                postInvalidate();
                this.b = motionEvent.getX();
            }
        }
        return true;
    }

    public void setMiddleLineColor(int i) {
        this.v0 = i;
        invalidate();
    }

    public void setScrollingListener(o27 o27) {
    }
}
